cmd/buildah/version.go

```diff
@@ -21,45 +22,86 @@ var (
 	cniVersion string
 )
 
-//Function to get and print info for version command
-func versionCmd(c *cobra.Command, args []string) error {
-	var err error
-	buildTime := int64(0)
-	if buildInfo != "" {
-		//converting unix time from string to int64
-		buildTime, err = strconv.ParseInt(buildInfo, 10, 64)
-		if err != nil {
-			return err
-		}
-	}
-
-	fmt.Println("Version:        ", define.Version)
-	fmt.Println("Go Version:     ", runtime.Version())
-	fmt.Println("Image Spec:     ", ispecs.Version)
-	fmt.Println("Runtime Spec:   ", rspecs.Version)
-	fmt.Println("CNI Spec:       ", cniversion.Current())
-	fmt.Println("libcni Version: ", cniVersion)
-	fmt.Println("image Version:  ", iversion.Version)
-	fmt.Println("Git Commit:     ", GitCommit)
-
-	//Prints out the build time in readable format
-	fmt.Println("Built:          ", time.Unix(buildTime, 0).Format(time.ANSIC))
-	fmt.Println("OS/Arch:        ", runtime.GOOS+"/"+runtime.GOARCH)
-
-	return nil
+type versionInfo struct {
+	Version       string `json:"version"`
+	GoVersion     string `json:"goVersion"`
+	ImageSpec     string `json:"imageSpec"`
+	RuntimeSpec   string `json:"runtimeSpec"`
+	CniSpec       string `json:"cniSpec"`
+	LibcniVersion string `json:"libcniVersion"`
+	ImageVersion  string `json:"imageVersion"`
+	GitCommit     string `json:"gitCommit"`
+	Built         string `json:"built"`
+	OsArch        string `json:"osArch"`
 }
 
-//cli command to print out the version info of buildah
-var versionCommand = &cobra.Command{
-	Use:     "version",
-	Short:   "Display the Buildah version information",
-	Long:    "Displays Buildah version information.",
-	RunE:    versionCmd,
-	Args:    cobra.NoArgs,
-	Example: `buildah version`,
+type versionOptions struct {
+	json bool
 }
 
 func init() {
+	var opts versionOptions
+
+	//cli command to print out the version info of buildah
+	versionCommand := &cobra.Command{
+		Use:   "version",
+		Short: "Display the Buildah version information",
+		Long:  "Displays Buildah version information.",
+		RunE: func(c *cobra.Command, args []string) error {
+			var err error
```

Comment:
Could you put this code in a separate function, So the init  is not as cluttered.

Reply:
Done!